### ISSUE TITLE
[Bug] Backups fail when backing up files with emoji characters

### ISSUE THREAD

**Issue by ioogithub** [NeoApplications/Neo-Backup#231]
**Description**
When an app has a file with an emoji character, the backup fails.

**Steps To Reproduce**
1. Save a file with a emoji character in an app
2. Backup that app
3. Errors Occured: BackupFailedException: java.io.IOException occured on data backup:
java.io.IOException: Entry 'files/{EMOJ} filename.mpl' closed at '0' before the '587788' bytes specified in the header were written.

**Expected behavior**
The app will be able to handle all valid filenames.

**System Information(please complete the following information):**
 - Device:  Moto G7Plus
 - Android Version: 10
 - ROM: LineageOS  
 - App's Version:  17

Note, these filesnames contain valid characters. All filemanagers and other apps work as expected with these filenames. The previous version (oandbackup) worked normally with these filenames so it must be a bug with the new code. There doesn't seem to be any way around this problem at the moment.

**Comment by ioogithub** [comment]
Update: I tested backing up this same app and files with 3.2.0 and it backed up and restored without any issues. The problem seems to be with the new 4.0.0 only.

I did some additional testing, along with the emoji characters, the apostrophe character: ' in a filename also throws a similar error.

**Comment by Iey4iej3** [comment]
I encounter a similar issue, at least the log looks similar: `closed at '0' before '27182' bytes specified in the header were written`. There seems no emoji in the filename, however.

**Comment by Iey4iej3** [comment]
The issue persists after having upgraded to `5.0.2 (5002)`, i.e., not fixed by https://github.com/machiav3lli/oandbackupx/pull/298

**Comment by machiav3lli** [comment]
@Iey4iej3 this has nothing to do with this issue. Your failed backup is resulting of the app you're trying to backup being open in the background and updating its data. Did you try to enable the STOP-CONT from advanced settings?

**Comment by Iey4iej3** [comment]
> @Iey4iej3 this has nothing to do with this issue. Your failed backup is resulting of the app you're trying to backup being open in the background and updating its data. Did you try to enable the STOP-CONT from advanced settings?

I am sure that I enabled it when I tested `5.0.0`. I will try for `5.0.2` later. Before that, I want to ask a simple question: this program has FCM enabled. I wonder whether it is sufficient to enable that setting? Maybe I will also test some kind of Task killer in addition to that advanced setting?

**Comment by Iey4iej3** [comment]
> @Iey4iej3 this has nothing to do with this issue. Your failed backup is resulting of the app you're trying to backup being open in the background and updating its data. Did you try to enable the STOP-CONT from advanced settings?

Now I have tested it relatively extensively. I tried to enable killing from advanced settings. The issue persists. Then I tried to kill the process in System Settings before backup, the issue still persists. I turned of "Start app on push message" in Cloud Messaging settings in `MicroG`, I get a weird mixed result: I still get IOException, but *sometimes* the backup file is generated (I don't know whether it is corrupt). From the name of the file, it does seem to be related to push notification.
